applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c

```diff
@@ -105,6 +105,98 @@ static struct carrier_op_list op_list[CARRIER_OP_MAX] = {
+	CARRIER_CFG_OP_BOOTSTRAP_FROM_SMARTCARD,
+	CARRIER_CFG_OP_CARRIERS,
+	CARRIER_CFG_OP_COAP_CON_INTERVAL,
+	CARRIER_CFG_OP_PDN_TYPE,
+	CARRIER_CFG_OP_SESSION_IDLE_TIMEOUT,
+	/* Device Custom Configuration */
+	CARRIER_CFG_OP_DEVICE_CONFIG_ENABLE,
+	CARRIER_CFG_OP_DEVICE_DEVICE_TYPE,
+	CARRIER_CFG_OP_DEVICE_HW_VERSION,
+	CARRIER_CFG_OP_DEVICE_MANUFACTURER,
+	CARRIER_CFG_OP_DEVICE_MODEL_NUMBER,
+	CARRIER_CFG_OP_DEVICE_SW_VERSION,
+	/* LG U+ Custom Configuration */
+	CARRIER_CFG_OP_LGU_DEVICE_SERIAL_NO_TYPE,
+	CARRIER_CFG_OP_LGU_SERVICE_CODE,
+	/* Server Custom Configuration */
+	CARRIER_CFG_OP_SERVER_CONFIG_ENABLE,
+	CARRIER_CFG_OP_SERVER_BINDING,
+	CARRIER_CFG_OP_SERVER_IS_BOOTSTRAP,
+	CARRIER_CFG_OP_SERVER_LIFETIME,
+	CARRIER_CFG_OP_SERVER_SEC_TAG,
+	CARRIER_CFG_OP_SERVER_URI,
+	/* Count */
+	CARRIER_CFG_OP_MAX
+};
+
+/** forward declaration of configuration cmd handlers **/
+static int do_cfg_apn(void);
+static int do_cfg_auto_startup(void);
+static int do_cfg_bootstrap_from_smartcard(void);
+static int do_cfg_carriers(void);
+static int do_cfg_coap_con_interval(void);
+static int do_cfg_device_enable(void);
+static int do_cfg_device_type(void);
+static int do_cfg_enable(void);
+static int do_cfg_hardware_version(void);
+static int do_cfg_manufacturer(void);
+static int do_cfg_model_number(void);
+static int do_cfg_session_idle_timeout(void);
+static int do_cfg_software_version(void);
+static int do_cfg_device_serial_no_type(void);
+static int do_cfg_service_code(void);
+static int do_cfg_pdn_type(void);
+static int do_cfg_binding(void);
+static int do_cfg_server_enable(void);
+static int do_cfg_is_bootstrap(void);
+static int do_cfg_lifetime(void);
+static int do_cfg_sec_tag(void);
+static int do_cfg_uri(void);
+
+/**@brief SLM Carrier Configuration AT Command list type. */
+static struct carrier_op_list cfg_op_list[CARRIER_CFG_OP_MAX] = {
+	{CARRIER_CFG_OP_APN, "apn", do_cfg_apn},
+	{CARRIER_CFG_OP_AUTO_STARTUP, "auto_startup", do_cfg_auto_startup},
+	{CARRIER_CFG_OP_BOOTSTRAP_FROM_SMARTCARD, "bootstrap_smartcard",
+	 do_cfg_bootstrap_from_smartcard},
+	{CARRIER_CFG_OP_CARRIERS, "carriers", do_cfg_carriers},
+	{CARRIER_CFG_OP_COAP_CON_INTERVAL, "coap_con_interval", do_cfg_coap_con_interval},
+	{CARRIER_CFG_OP_DEVICE_CONFIG_ENABLE, "device_enable", do_cfg_device_enable},
+	{CARRIER_CFG_OP_DEVICE_DEVICE_TYPE, "device_type", do_cfg_device_type},
+	{CARRIER_CFG_OP_DEVICE_HW_VERSION, "hardware_version", do_cfg_hardware_version},
+	{CARRIER_CFG_OP_DEVICE_MANUFACTURER, "manufacturer", do_cfg_manufacturer},
+	{CARRIER_CFG_OP_DEVICE_MODEL_NUMBER, "model_number", do_cfg_model_number},
+	{CARRIER_CFG_OP_DEVICE_SW_VERSION, "software_version", do_cfg_software_version},
+	{CARRIER_CFG_OP_ENABLE, "enable", do_cfg_enable},
+	{CARRIER_CFG_OP_LGU_DEVICE_SERIAL_NO_TYPE, "device_serial_no_type",
+	 do_cfg_device_serial_no_type},
+	{CARRIER_CFG_OP_LGU_SERVICE_CODE, "service_code", do_cfg_service_code},
+	{CARRIER_CFG_OP_PDN_TYPE, "pdn_type", do_cfg_pdn_type},
```

Comment:
This one and below used to be all CONFIG_LWM2M_CARRIER_xxxx items.
There is a need to check the new "nrf\lib\bin\lwm2m_carrier\Kconfig" together with this PR, I think

Reply:
As I mentioned in [this comment](https://github.com/nrfconnect/sdk-nrf/pull/13165#discussion_r1401633569), these are meant to allow the users to change the configurations at run-time. You can still choose to use Kconfig if the corresponding config group is disabled.